### ISSUE TITLE
Razor fails to properly format nested C#/Components inside of other components

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/aspnetcore#30382]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity2.visualstudio.com/t/Razor-file-indentation-problem/1344867)._

---
[severity:It bothers me. A fix would be nice] [regression] [worked-in:16.9.0 preview 4]
Hi,

I am using the new experimental Razor editor ( with the Razor Language Server Protocol ).

I have an issue with an incorrect formatting when I do "Format Document".

I provided a screenshot of the resulting document. Before the formatting, the "if" are correctly formatted like so:

```
@if(SelectedTab == "Active")
{
     <UsersTable>...
}
```
After the formatting, the <UserTable> is pulled back and the first '{' is tabbed :

```
@if(SelectedTab == "Active")
     {
<UsersTable>...
}
```
I have tested it somewhere else in my code to be sure, and got some different results by adding a comment at the end of the "if". For both test, I formatted it correctly first, then screenshot after it mess it up :

! [Screenshot 2021-02-20 235232.png] (https://aka.ms/dc/image?name=Bb46f09ce570d4efbbe0fbcf8580a2956637494800140628813_Screenshot_2021-02-20_235232.png&tid=b46f09ce570d4efbbe0fbcf8580a2956637494800140628813)

! [Screenshot 2021-02-20 235425.png] (https://aka.ms/dc/image?name=B5f678bd62d8f460ca673db982982d8ea637494800738417944_Screenshot_2021-02-20_235425.png&tid=5f678bd62d8f460ca673db982982d8ea637494800738417944)

---
#### Feedback Bot on 2/21/2021, 10:27 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by GerkinDev** [comment]
Reproduced with a foreach loop too: 

```razor
<SomeFooContainer>
    @foreach (var item in items)
    {
        <SomeFoo>@item</SomeFoo>
    }
</SomeFooContainer>
```

gets formatted to


```razor
<SomeFooContainer>
    @foreach (var item in items)
            {
        <SomeFoo>@item</SomeFoo>
    }
</SomeFooContainer>
```

This is still a huge improvement compared to the previous razor formatter. Thanks for your work on it. 🥂

**Comment by davidwengier** [comment]
Also reported in https://developercommunity.visualstudio.com/t/razor-formater-work-not-correct/1434085

**Comment by davidwengier** [comment]
Notes for me, or anyone who is curious:

We get a series of text edits from C# formatting which could be for outside of out mapping. So for example, given this code (dots used to show spaces):

```
·················if (true)
················{
················

#line default
#line hidden
#nullable disable
#nullable restore
#line 15 "/path/to/Document.component"
·································
················}
```

We might be asking for a format from the end of the 2nd last line, to the end of the brace, but C# formatting will give us back edits to remove all of the whitespace after the open brace too. We only want the edits around the close brace, except in this case C# will give us back a single edit for removing the whitespace on the second last line, and fixing the indentation of the close brace. Unfortunately since the whitespace on the second last line starts before our mapping, we consider the edit unmapped and throw it away.

**Comment by davidwengier** [comment]
hmm.. now that I've typed that up, I'm not sure if thats the bug. I'm leaving it for posterity anyway, it might come in handy later.

**Comment by b-straub** [comment]
Still happens in VS Version 16.11.0 Preview 3.0, very annoying.

**Comment by b-straub** [comment]
Fixed in VS Version 17.0.0 Preview 2.0 but not in Version 16.11.0 Preview 3.0!
